Assets/Global/Scripts/UIScripts/Healthbar.cs

```diff
@@ -4,19 +4,40 @@
 
 public class Healthbar : MonoBehaviour
 {
-    private Image healthbarOverlay;
-    private Image healthbarUnderlay;
+    private Transform HealthbarUnderlayTransform;
+    private Transform HealthbarOverlayTransform;
+    private float originalWidth;
     // Start is called once before the first execution of Update after the MonoBehaviour is created
-    void Start()
+    void Awake()
     {
-        healthbarOverlay = transform.GetChild(1).GetComponent<Image>();
-        healthbarUnderlay = transform.GetChild(0).GetComponent<Image>();
+        HealthbarUnderlayTransform = transform.GetChild(0);
+        HealthbarOverlayTransform = transform.GetChild(1);
```

Comment:
Instead of parsing  `Transform` to `RectTransform` everywhere. Just parse it where ever you retrieve the object. 
Since now you are parsing it on literally every line you are using it. And so it would be a lot more readable if you have like 
`private RectTransform HealthbarUnderlayTransform;`
and then
`HealthbarUnderlayTransform = transform.GetChild(0) as RectTransform;`

(Note that RectTransform is also a Transform, so you wont lose any functionality by just saving it as a RectTransform instead)